src/drive/web/modules/queries.js

```diff
@@ -77,6 +77,32 @@ const buildRecentQuery = () => ({
   }
 })
 
+/**
+ * Returns one file with specific metadata for Recent view
+ * Only one file is necessary because it allows us to know whether or not to display
+ * the column for this specific metadata (like carbonCopy or electronicSafe).
+ * @param {object} params - Params
+ * @param {string} params.attribute - Metadata attribute
+ */
+export const buildRecentWithMetadataAttributeQuery = ({ attribute }) => ({
+  definition: () =>
+    Q('io.cozy.files')
+      .where({
+        type: 'file',
+        trashed: false,
+        [`metadata.${attribute}`]: true,
+        updated_at: {
+          $gt: null
+        }
+      })
+      .indexFields([`metadata.${attribute}`, 'type', 'trashed', 'updated_at'])
+      .limitBy(1),
```

Comment:
Nit: If we want to clearly show that buildRecentWithMetadata is the same as buildRecent with more stuff, we could base ourselves on the buildRecentQuery

```
const q = buildRecentQuery()
q.where({ ...q.selector, [`metadata.${attribute}`]: true })
  .indexFields(...q.indexedFields, )
```

Reply:
Oui je pense que ça peut être une bonne amélioration par la suite en effet. Je le note dans le code pour pas l'oublier.